### PR TITLE
REFACTOR-#4284: use variable length unpacking when getting results from `deploy` function

### LINKED ISSUE #4284: REFACTOR: use variable length unpacking when getting results from `deploy` function
Initial found: https://github.com/modin-project/modin/pull/3441#discussion_r815934541

[Link](https://www.python.org/dev/peps/pep-3132/) to python docs.

Example:
```python
### BEFORE
partition_id = cls.deploy(cls.parse, num_splits + 3, args)
partition_ids.append(partition_id[:-3])
index_ids.append(partition_id[-3])
dtypes_ids.append(partition_id[-2])
### AFTER
*partition_ids[idx], index_ids[idx], dtypes_ids[idx] = cls.deploy(
```

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Anatoly Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4284 
- [x] tests ~added~ and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

File: modin/core/io/text/json_dispatcher.py [hunk 1]
```diff
@@ -57,25 +57,20 @@ def _read(cls, path_or_buf, **kwargs):
...
+            index_ids = [None] * len(splits)
+            dtypes_ids = [None] * len(splits)
+            for idx, (start, end) in enumerate(splits):
                 args.update({"start": start, "end": end})
-                partition_id = cls.deploy(cls.parse, num_splits + 3, args)
-                partition_ids.append(partition_id[:-3])
-                index_ids.append(partition_id[-3])
-                dtypes_ids.append(partition_id[-2])
-
+                *partition_ids[idx], index_ids[idx], dtypes_ids[idx], _ = cls.deploy(
```
Comment (devin-petersohn) [inline modin/core/io/text/json_dispatcher.py:71]: Why the extra `_`? Is that value unused?

**Review by devin-petersohn — COMMENTED** [review]
Seems fine to me to change this, left a question.

File: modin/core/io/text/json_dispatcher.py (hunk 1 above)
Comment (anmyachev) [reply modin/core/io/text/json_dispatcher.py:71]: `PandasJSONParser` returns [also](https://github.com/modin-project/modin/blob/7c02bafa1badade0ba9d1916652ded74bc36a1b9/modin/core/storage_formats/pandas/parsers.py#L625) `partitions_columns` value. This is laid down for the future - to implement `lines=False` case. (According to [the comment](https://github.com/modin-project/modin/blob/7c02bafa1badade0ba9d1916652ded74bc36a1b9/modin/core/io/text/json_dispatcher.py#L79))

**Comment by anmyachev** [comment]
@devin-petersohn can we merge the PR?

**Review by devin-petersohn — APPROVED** [review]
lgtm!

File: docs/release_notes/release_notes-0.14.0.rst [hunk 2]
```diff
@@ -44,6 +44,7 @@ Key Features and Updates
   * REFACTOR-#4206: add assert check into `__init__` method of `PandasOnDaskDataframePartition` class (#4207)
   * REFACTOR-#3900: add flake8-no-implicit-concat plugin and refactor flake8 error codes (#3901)
   * REFACTOR-#4093: Refactor base to be smaller (#4220)
+  * REFACTOR-#4284: use variable length unpacking when getting results from `deploy` function (#4285)
```
Comment (YarShev) [inline docs/release_notes/release_notes-0.14.0.rst:47]: This should be moved in `docs/release_notes/release_notes-0.15.0.rst`.

Comment (YarShev) [reply docs/release_notes/release_notes-0.14.0.rst:47]: @anmyachev, when adding this in the file please add also your GH nickname in `Contributors` list in there.

Comment (anmyachev) [reply docs/release_notes/release_notes-0.14.0.rst:47]: https://github.com/modin-project/modin/pull/4403

Comment (YarShev) [reply docs/release_notes/release_notes-0.14.0.rst:47]: @anmyachev, thanks!

**Pull request merged**